group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupCoordinatorRecordSerde.java

```diff
@@ -53,90 +49,11 @@
 import org.apache.kafka.coordinator.group.generated.ShareGroupTargetAssignmentMemberValue;
 import org.apache.kafka.coordinator.group.generated.ShareGroupTargetAssignmentMetadataKey;
 import org.apache.kafka.coordinator.group.generated.ShareGroupTargetAssignmentMetadataValue;
-import org.apache.kafka.server.common.ApiMessageAndVersion;
 
-import java.nio.BufferUnderflowException;
-import java.nio.ByteBuffer;
-
-/**
- * Serializer/Deserializer for {@link CoordinatorRecord}. The format is defined below:
- * <pre>
- *     record_key   = [record_type key_message]
- *     record_value = [value_version value_message]
- *
- *     record_type     : The record type is currently define as the version of the key
- *                       {@link ApiMessageAndVersion} object.
- *     key_message     : The serialized message of the key {@link ApiMessageAndVersion} object.
- *     value_version   : The value version is currently define as the version of the value
- *                       {@link ApiMessageAndVersion} object.
- *     value_message   : The serialized message of the value {@link ApiMessageAndVersion} object.
- * </pre>
- */
 @SuppressWarnings({ "ClassDataAbstractionCoupling", "CyclomaticComplexity" })
```

Comment:
Do we still need these warnings after refactoring?

Reply:
Yes
```
[ant:checkstyle] [ERROR] /home/smjn/ws/kafka/group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupCoordinatorRecordSerde.java:54:1: Class Data Abstraction Coupling is 31 (max allowed is 25) 
...
[ClassDataAbstractionCoupling]
[ant:checkstyle] [ERROR] /home/smjn/ws/kafka/group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupCoordinatorRecordSerde.java:55:5: Cyclomatic Complexity is 17 (max allowed is 16). [CyclomaticComplexity]
[ant:checkstyle] [ERROR] /home/smjn/ws/kafka/group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupCoordinatorRecordSerde.java:94:5: Cyclomatic Complexity is 17 (max allowed is 16). [CyclomaticComplexity]
```

Reply:
There is a better way to suppress checkstyle warnings. Look at "checkstyle/suppressions.xml"